crates/pathfinder/src/rpc/types.rs

```diff
@@ -92,6 +92,30 @@ pub mod request {
             BlockResponseScope::TransactionHashes
         }
     }
+
+    /// Contains event filter parameters passed to `starknet_getEvents`.
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+    #[serde(deny_unknown_fields)]
+    pub struct EventFilter {
+        #[serde(default, rename = "fromBlock")]
+        pub from_block: Option<StarknetBlockNumber>,
+        #[serde(default, rename = "toBlock")]
+        pub to_block: Option<StarknetBlockNumber>,
+        #[serde(default)]
+        pub address: Option<ContractAddress>,
+        #[serde(default)]
+        pub keys: Vec<EventKey>,
+
+        #[serde(flatten)]
+        pub pagination: Option<ResultPageRequest>,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+    #[serde(deny_unknown_fields)]
```

Comment:
Maybe trying to compose the request from two types isn't a good idea anyway.
I mean I guess the specification seems to be doing schema composition incorrectly with `allOf`, and the construct above was only my try to mirror what's in the specs. But since the paging request is only used here we should maybe just include the `page_size` and `page_number` fields to `EventFilter`?